lib/spack/spack/spec.py

```diff
@@ -5245,10 +5272,18 @@ def spec(self, name):
                 vlist = self.version_list()
                 spec._add_versions(vlist)
 
+            elif self.accept(D_ON):
+                name = self.variant()
+                spec.variants[name] = vt.BoolValuedVariant(name, True, True)
```

Comment:
```suggestion
                spec.variants[name] = vt.BoolValuedVariant(name, True, propagate=True)
```
Just a little easier to read this way